src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -396,37 +418,28 @@ private CompressionMethodValues CompressionMethod
             }
         }
 
-        private string DecodeEntryName(byte[] entryNameBytes)
+        // Converts the specified bytes into a string of the proper encoding for this ZipArchiveEntry.
+        private string DecodeBytesToString(byte[] bytes)
```

Comment:
This method should be re-used for decoding the new properties. You can make it static and replace the instance fields that uses with parameters, that way you can use it with ZipArchive and ZipArchiveEntry.

Reply:
Actually, this method currently is only used in the constructor to set the value of the `FullName` property. I think I can simplify the code by removing the method.

The logic of `FullName` can further be improved by making it look more like the current logic for `Comment` (mainly, by removing the `string`, and keeping just the `byte[]`). But I'd like to do that in a separate PR if you don't mind.

Reply:
Or maybe I can submit this in a single commit, hopefully it won't complicate the current PR too much. Let me try. If we don't want to have it in the same PR, I can remove it. cc @adamsitnik